# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoadRequest.h
# Commit: 0a3d9301244b
# Full Hash: 0a3d9301244b91559c4c19314ce991525d9f605c
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move mElement from ScriptFetchOptions to DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D136742
# ==============================================================================

diff -r 46d45c7bae2f -r 0a3d9301244b dom/script/ModuleLoadRequest.h
--- a/dom/script/ModuleLoadRequest.h	Fri Feb 18 12:45:48 2022 +0000
+++ b/dom/script/ModuleLoadRequest.h	Fri Feb 18 12:45:49 2022 +0000
@@ -51,11 +51,9 @@
                                                          ScriptLoadRequest)
 
   // Create a top-level module load request.
-  static ModuleLoadRequest* CreateTopLevel(nsIURI* aURI,
-                                           ScriptFetchOptions* aFetchOptions,
-                                           const SRIMetadata& aIntegrity,
-                                           nsIURI* aReferrer,
-                                           ScriptLoader* aLoader);
+  static ModuleLoadRequest* CreateTopLevel(
+      nsIURI* aURI, ScriptFetchOptions* aFetchOptions, Element* aElement,
+      const SRIMetadata& aIntegrity, nsIURI* aReferrer, ScriptLoader* aLoader);
 
   // Create a module load request for a static module import.
   static ModuleLoadRequest* CreateStaticImport(nsIURI* aURI,
@@ -64,7 +62,8 @@
   // Create a module load request for dynamic module import.
   static ModuleLoadRequest* CreateDynamicImport(
       nsIURI* aURI, ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
-      ScriptLoader* aLoader, JS::Handle<JS::Value> aReferencingPrivate,
+      Element* aElement, ScriptLoader* aLoader,
+      JS::Handle<JS::Value> aReferencingPrivate,
       JS::Handle<JSString*> aSpecifier, JS::Handle<JSObject*> aPromise);
 
   bool IsTopLevel() const override { return mIsTopLevel; }